File: web-app/src/app/services/types/Skill.ts
import { JobpostingSkill } from "./JobpostingSkill";
import { SkillType } from "./SkillType";
import { StudentSkill } from "./StudentSkill";

export type Skill = {
  Id: number;
  Name?: string;
  SkillTypeId?: number;
  SkillType?: SkillType;
  JobpostingSkills?: JobpostingSkill[];
  StudentSkills?: StudentSkill[];
};

File: Placements.DataAccess/Placement/Models/Skill.cs
using System;
using System.Collections.Generic;

namespace Placements.DataAccess.Placement.Models;

public partial class Skill
{
    public long Id { get; set; }

    public string? Name { get; set; }

    public int? SkillTypeId { get; set; }

    public virtual ICollection<JobpostingSkill> JobpostingSkills { get; set; } = new List<JobpostingSkill>();

    public virtual SkillType? SkillType { get; set; }

    public virtual ICollection<StudentSkill> StudentSkills { get; set; } = new List<StudentSkill>();
}
